modules/nw-allocate-load-balancers-to-subnets-procedure.adoc

```diff
@@ -0,0 +1,124 @@
++
+For each AWS subnet you intend the cluster to use, you will create an entry specifying its `id` and a list of `roles`. Each role is an object with a `type` key. To designate a subnet for the default Ingress Controller, assign it a role with `type: IngressControllerLB`.
++
+[source,yaml]
+----
+apiVersion: v1
+baseDomain: example.com <1>
+metadata:
+  name: my-cluster # Example cluster name
+platform:
+  aws:
+    region: us-east-1 <2>
+    vpc: <3>
+      subnets: <4>
+      - id: subnet-0fcf8e0392f0910d5 # Public Subnet in AZ us-east-1a <5>
+        roles:
+        - type: IngressControllerLB <6>
+        - type: BootstrapNode
+      - id: subnet-0xxxxxxxxxxxxxxza # Public Subnet in another AZ for HA
+        roles:
+        - type: IngressControllerLB
+      - id: subnet-0fcf8e0392f0910d4 # Private Subnet in AZ us-east-1a
+        roles:
+        - type: ClusterNode <7>
+      - id: subnet-0yyyyyyyyyyyyyyzb # Private Subnet in another AZ for HA
+        roles:
+        - type: ClusterNode
+      # Add other subnet IDs and their roles as needed for your cluster architecture
+pullSecret: '...' <8>
+sshKey: '...' <9>
+----
+<1> Your base domain.
+<2> Your AWS region.
+<3> The vpc object under platform.aws contains the subnets list.
+<4> List of all subnet objects that OpenShift will use. Each object defines a subnet id and its roles.
+<5> Replace with your actual AWS Subnet ID. 
+<6> The `type: IngressControllerLB` role specifically designates this subnet for the default Ingress Controller's LoadBalancer. In private/internal cluster, the subnet with `IngressControllerLB` role must be private.
+<7> The `type: ClusterNode` role designates this subnet for control plane and compute nodes. These are typically private subnets.
+<8> Your pull secret.
+<9> Your SSH key.
++
+Entries for Control Plane load balancers in the `subnets` list would follow a similar pattern:
++
+[source,yaml]
+----
+# ... (within platform.aws.vpc.subnets list)
+      - id: subnet-0fcf8e0392f0910d6 # Public Subnet for External API LB
+        roles:
+        - type: ControlPlaneExternalLB
+      - id: subnet-0fcf8e0392f0910d7 # Private Subnet for Internal API LB
+        roles:
+        - type: ControlPlaneInternalLB
+# ...
+----
++
+For the default public Ingress Controller, any subnet assigned the `IngressControllerLB` role in your `install-config.yaml` file must be a public subnet. For example, it must have a route table entry in AWS that directs outbound traffic to an internet gateway (IGW).
++
+Ensure you list all necessary subnets, public and private across the AZs, and assign them appropriate roles according to your cluster architecture. 
++
+Subnet IDs define the subnets in an existing VPC and can optionally specify their intended roles. If no roles are specified on any subnet, the subnet roles are decided automatically. In this case, the VPC must not contain any other non-cluster subnets without the `kubernetes.io/cluster/` tag.
```

Comment:
nit: `kubernetes.io/cluster/` is a prefix. The correct tag key is `kubernetes.io/cluster/<cluster-id>`.

We can either say: `without a tag with prefix...` or `without the kubernetes.io/cluster/<cluster-id> tag`.